File: angular16-boilerplate/src/app/shared/models/agreement.ts
export class Agreement {
    public id: number;
    public companyId: number;
    public startDate: Date;
    public endDate: Date;
    public description?: string | null;
    public floor?: string | null;
    public officeNumber?: string | null;
    public section?: string | null;
    public workStation?: string | null;
  
    constructor(
      id: number,
      companyId: number,
      startDate: Date,
      endDate: Date,
      description?: string | null,
      floor?: string | null,
      officeNumber?: string | null,
      section?: string | null,
      workStation?: string | null
    ) {
      this.id = id;
      this.companyId = companyId;
      this.startDate = startDate;
      this.endDate = endDate;
      this.description = description;
      this.floor = floor;
      this.officeNumber = officeNumber;
      this.section = section;
      this.workStation = workStation;
    }
  }

File: webapi/DAL/Models/Agreement.cs
using System;
using System.Collections.Generic;

namespace DAL.Models
{
    public partial class Agreement
    {
        public Agreement()
        {
            Payments = new HashSet<Payment>();
        }

        public int Id { get; set; }
        public int CompanyId { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string? Description { get; set; }
        public string? Floor { get; set; }
        public string? OfficeNumber { get; set; }
        public string? Section { get; set; }
        public string? WorkStation { get; set; }
        public string? Branch { get; set; }

        public virtual Company Company { get; set; } = null!;
        public virtual ICollection<Payment> Payments { get; set; }
    }
}
